samples/java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartInterface2.java

```diff
@@ -0,0 +1,51 @@
+package com.example.shoppingcart.domain;
+
+import com.akkaserverless.javasdk.Effect;
+import com.akkaserverless.javasdk.Reply;
+import com.akkaserverless.javasdk.eventsourcedentity.CommandContext;
+import com.akkaserverless.javasdk.eventsourcedentity.CommandHandler;
+import com.akkaserverless.javasdk.eventsourcedentity.EventHandler;
+import com.akkaserverless.javasdk.eventsourcedentity.EventSourcedEntityEffect;
+import com.akkaserverless.javasdk.eventsourcedentity.Snapshot;
+import com.akkaserverless.javasdk.eventsourcedentity.SnapshotHandler;
+import com.example.shoppingcart.ShoppingCartApi;
+import com.google.protobuf.Empty;
+
+/** An event sourced entity. */
+public abstract class ShoppingCartInterface2 {
+    
+    public class CommandNotImplementedException extends UnsupportedOperationException {
+        public CommandNotImplementedException() {
+            super("You have either created a new command or removed the handling of an existing command. Please declare a method in your \"impl\" class for this command.");
+        }
+    }
+
+    protected abstract ShoppingCartDomain.Cart emptyState();
+
+    @CommandHandler(name = "AddItem")
+    public abstract Effect<Empty> addItem(
+            ShoppingCartApi.AddLineItem command,
+            ShoppingCartDomain.Cart currentState,
+            EventSourcedEntityEffect.Builder<Empty, ShoppingCartDomain.Cart> effectBuilder,
+            CommandContext ctx);
+
+    @CommandHandler(name = "RemoveItem")
+    public abstract Effect<Empty> removeItem(
+            ShoppingCartApi.RemoveLineItem command,
+            ShoppingCartDomain.Cart currentState,
+            EventSourcedEntityEffect.Builder<Empty, ShoppingCartDomain.Cart> effectBuilder,
+            CommandContext ctx);
```

Comment:
what will be left in the command context? 

Otherwise, we could have a `CallContext<State>` that we use to get command context and build the reply. 

(CallContext sucks, I just need to pick a name.)

Reply:
Combining into one context / effect builder would be good. And a simpler name too. Does look too heavy currently.

Reply:
I have included the effect builder in the CommandContext: fc680a3

Reply:
👍🏼 looks better behind the CommandContext.

Small thing, but I wonder if it would be cleaner as just `effects` rather than `effectBuilder`:

```java
return context.effects()
        .emitEvent(event)
        .thenReply(newState -> Empty.getDefaultInstance());
```

but maybe it's clearer to say that it's a builder explicitly.

Reply:
I like the suggestion of `context.effects()`.

On that topic. Effect is typically well know terminology for those that know some functional programming, but is it known for the target Java developers? Do we have any alternative naming suggestion?

Action? but that would be confusing since we already have that for Action components.

Just looking at the english dictionary for Effect, it might be just fine.

Reply:
I think Effect is fine, and could be useful having it aligned with Akka.

Agree that Action would be confusing. Alternative could be something like Operation.